test/Network/ConcurrencySpec.hs

```diff
@@ -0,0 +1,36 @@
+{-# LANGUAGE OverloadedStrings #-}
+module Network.ConcurrencySpec where
+
+import Test.Hspec
+import qualified Network.Freddy as Freddy
+import qualified Network.Freddy.Request as R
+import Control.Concurrent.Async (forConcurrently)
+import System.Timeout (timeout)
+import Data.Maybe (isJust)
+import SpecHelper (
+  randomQueueName,
+  requestBody,
+  withConnection,
+  delayedResponder
+  )
+
+spec :: Spec
+spec = around withConnection $
+  describe "Concurrency" $ do
+    let buildRequest queueName = R.newReq {
+      R.queueName = queueName,
+      R.body = requestBody
+    }
+
+    it "runs responder in parallel" $ \connection -> do
+      queueName <- randomQueueName
+
+      Freddy.respondTo connection queueName $ delayedResponder 10
+
+      result <- timeout (20 * 1000) $
+        forConcurrently [1, 2, 3, 4] $ \n ->
```

Comment:
Even though `forConcurrently` creates a thread the `replyCallback` is still ran in the consumer channel thread without functionality changes.